osu.Game/Overlays/BeatmapSet/Header.cs

```diff
@@ -208,19 +215,23 @@ public Header()
             {
                 Details.Beatmap = b.NewValue;
                 externalLink.Link = $@"https://osu.ppy.sh/beatmapsets/{BeatmapSet.Value?.OnlineBeatmapSetID}#{b.NewValue?.Ruleset.ShortName}/{b.NewValue?.OnlineBeatmapID}";
+
+                tabControl.SelectRuleset(b.NewValue?.Ruleset);
```

Comment:
This should never be the case - the tabcontrol never changes if the beatmap changes.